neural/neural_discern_run_script.py

```diff
@@ -0,0 +1,345 @@
+        max_val, max_indx = torch.topk(attnw, topk, dim=1)
+        print(docid)
+        print("attended sent:")
+        for i in range(max_indx.size(-1)):
+            target_indx = max_indx[0][i].item()
+            print("sentence num:", target_indx, "attnw:", max_val[0][i].item())
+            print(proc_articles_repr[docid]['sents'][target_indx])
+        print()
+
+
+def verbose_print(text, print_yn):
+    if print_yn:
+        print(text)
+
+# ============================================================
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--test-mode", action="store_true", default=False, help="run in abbreviated test mode")
+    parser.add_argument("--biobert", default=True, help="Use the BioBERT embeddings, or the normal BERT")
+    parser.add_argument("--rewrite-sentence-embeddings", action="store_true", default=False, help="(Re-)compute the "
+                                                                                                  "sentence embeddings")
+    parser.add_argument("--run-hyper-param-search", default=True, help="Run the hyper parameter search")
+    parser.add_argument("--hyperparam-search-dir", default=None, help="Re-use the results of a pre-run hyperparam "
+                                                                      "search")
+    parser.add_argument("--base-dir", default='/opt/data/autodiscern/aa_neural', help="Base dir to and including "
+                                                                                      "autodiscern/aa_neural/")
+    args = parser.parse_args()
+
+    config = {
+        'test_mode': args.test_mode,
+        'biobert': args.biobert,
+        'rewrite_sentence_embeddings': args.rewrite_sentence_embeddings,
+        'run_hyper_param_search': args.run_hyper_param_search,
+        'hyperparam_search_dir': args.hyperparam_search_dir,
+        'questions_to_run': [4, 5, 9, 10, 11],
+        'max_folds': 5,
+        'num_epochs': 25,
+        'verbose': True,
+        'experiment_to_rerun': None,
+        'questions': (4, 5, 9, 10, 11),
+        'question_gpu_map': {4: 1, 5: 2, 9: 3, 10: 4, 11: 5},
+        'base_dir': args.base_dir,
+    }
+
+    if config['hyperparam_search_dir'] and config['run_hyper_param_search']:
+        print("WARNING: you selected a hyperparam search dir while also setting run-hyper-param-search as True. "
+              "The pre-built hyperparam search dir will be used. Hyperparam search will not be run.")
+
+    # under test mode (for faster debugging), run a smaller set of partitions and epochs, and no hyper param search
+    if config['test_mode']:
+        config['max_folds'] = 2  # max number of data partition folds to run (for faster testing)
+        config['num_epochs'] = 2
+        config['run_hyper_param_search'] = False
+
+    time_stamp = datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+    if config['experiment_to_rerun']:
```

Comment:
I kind of follow the logic here but not quite sure how can you set the `copy_exp_dir` in `argparser`. Anyhow, I am good here

Reply:
Added this to the config, and added official command line args for this and `rerun-experiment-dir` (FYI I'm using these args to re-run the official bert run as I add infrastructure to also save predictions in the results).